File: hs/codewars/reverse_every_other_word_in_the_string.hs
-- https://www.codewars.com/kata/reverse-every-other-word-in-the-string

module ReverseEveryOtherWord (reverseEveryOther) where

import Data.List
import Data.List.Split

reverseIfOdd :: Int -> [Char] -> [Char]
reverseIfOdd i s
  | i `mod` 2 == 0 = s
  | otherwise = reverse s

reverseEveryOther :: String -> String
reverseEveryOther s = intercalate " "
  $ zipWith reverseIfOdd [0..]
  $ splitOn " " s

File: scala/codewars/reverse_every_other_word_in_the_string.scala
// https://www.codewars.com/kata/reverse-every-other-word-in-the-string

object Solution {
  def reverse(s: String): String = {
    (s.split(" ").zipWithIndex
      map {case (x, i) => if (i % 2 == 0) x else x.reverse}
      mkString " ")
  }
}
